test/ops/random.jl

```diff
@@ -24,25 +25,53 @@
         block = _eval(n)
         @test block.times == expected_times
         @test all(typemin(Int32) .<= block.values .<= typemax(Int32))
+
+        # No explicit rng.
+        n = rand(x, Int32)
+        @test n != rand(x, Int32)
+        @test value_type(n) == Int32
+
+        @testset "collections of values" begin
+            for vals in (
+                (1, 2.0, 2.0 + 0im),
+                [1, 2],
+                [1 2; 3 4],
+                Set([1, 2, 3]),
+                Dict(1 => "1", 2 => "2", 3 => "3"),
+            )
+                n = rand(rng, x, vals)
+                @test n === rand(rng, x, vals)
+                @test value_type(n) <: foldl(Base.promote_typejoin, typeof.(collect(vals)))
```

Comment:
Check that `value_type(n)` is a SUPERTYPE of every element in vals.
Same below in array case.